app/actions/ModActionPanel/QuickAction.tsx

```diff
@@ -828,22 +834,37 @@ function Form(
 function useSubjectQuery(subject: string) {
```

Comment:
nit, and probably a more general comment, but I think that the get profile part would be better implemented as its own `useSubjectProfile` hook. This would allow for better composability and maintainability in the future.

Reply:
yeah that's fair. just don't have many places here where profile is needed in isolation so feels more natural for me to tie it to repo/record data fetcher for now at least.